tests/withBlockTracker.ts

```diff
@@ -0,0 +1,252 @@
+          callback(null as unknown as Error, {
+            jsonrpc: '2.0',
+            id: 1,
+            error: {
+              code: -999,
+              message: stub.response.error,
+            },
+          });
+        }
+      } else if ('error' in stub) {
+        callback(new Error(stub.error), null as unknown as JsonRpcResponse<U>);
+      }
+      return;
+    }
+
+    callback(
+      new Error(
+        `Could not find any stubs matching "${request.method}". Perhaps they've already been called?\n\n` +
+          'The original set of stubs were:\n\n' +
+          `${util.inspect(this.originalStubs, { depth: null })}\n\n` +
+          'Current set of stubs:\n\n' +
+          `${util.inspect(this.stubs, { depth: null })}\n\n`,
+      ),
+      null as unknown as JsonRpcResponse<U>,
+    );
+  }
+
+  private buildStubsFrom(givenStubs: FakeProviderStub[]): FakeProviderStub[] {
+    const stubs = givenStubs.slice();
+
+    if (!stubs.some((stub) => stub.methodName === 'eth_blockNumber')) {
+      stubs.push({
+        methodName: 'eth_blockNumber',
+        response: {
+          result: '0x0',
+        },
+      });
+    }
+
+    if (!stubs.some((stub) => stub.methodName === 'eth_subscribe')) {
+      stubs.push({
+        methodName: 'eth_subscribe',
+        response: {
+          result: '0x0',
+        },
+      });
+    }
+
+    if (!stubs.some((stub) => stub.methodName === 'eth_unsubscribe')) {
+      stubs.push({
+        methodName: 'eth_unsubscribe',
+        response: {
+          result: true,
+        },
+      });
+    }
+
+    return stubs;
+  }
+}
+
+/**
+ * Calls the given function with a built-in PollingBlockTracker, ensuring that
+ * all listeners that are on the block tracker are removed and any timers or
+ * loops that are running within the block tracker are properly stopped.
+ *
+ * @param options - Options that allow configuring the block tracker or
+ * provider.
+ * @param callback - A callback which will be called with the built block
+ * tracker.
+ */
+async function withPollingBlockTracker(
+  options: WithPollingBlockTrackerTypes['withBlockTrackerOptions'],
+  callback: WithPollingBlockTrackerTypes['withBlockTrackerCallback'],
+): Promise<void>;
+/**
+ * Calls the given function with a built-in PollingBlockTracker, ensuring that
+ * all listeners that are on the block tracker are removed and any timers or
+ * loops that are running within the block tracker are properly stopped.
+ *
+ * @param callback - A callback which will be called with the built block
+ * tracker.
+ */
+async function withPollingBlockTracker(
+  callback: WithPollingBlockTrackerTypes['withBlockTrackerCallback'],
+): Promise<void>;
+/* eslint-disable-next-line jsdoc/require-jsdoc */
+async function withPollingBlockTracker(...args: any[]) {
+  const callback: WithPollingBlockTrackerTypes['withBlockTrackerCallback'] =
+    args.pop();
+  const options =
+    (args[0] as WithPollingBlockTrackerTypes['withBlockTrackerOptions']) ?? {};
```

Comment:
We could avoid using `as` here by including the full argument types. It doesn't seem to interfere with the overrides being picked up by VSCode.

e.g.
```suggestion
async function withPollingBlockTracker(
  ...args:
    | [
        WithPollingBlockTrackerTypes['withBlockTrackerOptions'],
        WithPollingBlockTrackerTypes['withBlockTrackerCallback'],
      ]
    | [WithPollingBlockTrackerTypes['withBlockTrackerCallback']]
) {
  let callback: WithPollingBlockTrackerTypes['withBlockTrackerCallback'];
  let options: WithPollingBlockTrackerTypes['withBlockTrackerOptions'] = {};

  if (args.length === 2) {
    options = args[0];
    callback = args[1];
  } else {
    callback = args[0];
  }
```

And the same could be done for the `withSubscribeBlockTracker` function

Reply:
Ahhh good point. I can't believe I didn't think of this. Fixed in [c3e20c9](https://github.com/MetaMask/eth-block-tracker/pull/103/commits/c3e20c989f39fad72cc1bc2f002839d6db4f61cd).